analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs

```diff
@@ -91,6 +114,14 @@ private static ProgramState ProcessArrayCreation(ProgramState state, IArrayCreat
     }
 
     private static bool IsCryptographicallyStrongRandomNumberGenerator(IInvocationOperationWrapper invocation) =>
-      (invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetBytes)) || invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetNonZeroBytes)))
-      && invocation.TargetMethod.ContainingType.DerivesFrom(KnownType.System_Security_Cryptography_RandomNumberGenerator);
+        (invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetBytes)) || invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetNonZeroBytes)))
+        && invocation.TargetMethod.ContainingType.DerivesFrom(KnownType.System_Security_Cryptography_RandomNumberGenerator);
+
+    // In C# the source collection for IEnumerable extension methods is stored in the argument named "source".
+    // In VB.NET that collection is the instance object of the invocation.
+    private static IOperation SourceArgumentOperation(IInvocationOperationWrapper invocation, ProgramState state) =>
```

Comment:
@mary-georgiou-sonarsource, @Tim-Pohlmann  
Should this be moved outside the class and made into a helper method? WDYT?
It's strange that extension methods are interpreted differently in C# and VB.
- C#: The caller instance (parameter marked with the `this `keyword) is the first argument. The `Instance` property of the invocation is `null`.
- VB: The caller instance is stored in the `Instance` property of the invocation and not found in the arguments.
Should this be handled by the SE engine in a more unified way?